ethtool update for legacy OS

juhlee-microsoft opened the pull request:
Found test failed in previous test cycle, and those tests; ETHTOOL-GET-FEATURES, ETHTOOL-GET-SET-CHANNEL, ETHTOOL-GET-SET-RINGBUFFER, ETHTOOL-GET-SET-TCP-HASHLEVELS, ETHTOOL-GET-SET-UDP-HASHLEVELS failed because they don't support those features. Inside it calls aborted instead of skipped. Like logging, it should be skipped.

LiliDeng commented:
Resolve #731

juhlee-microsoft commented:
@Tong-Xu @LiliDeng Please help me understand why you worked on cosmetic stuff even there are multiple P0 and P1 issues open. Why is it necessary to rename the variable?

LiliDeng commented:
@juhlee-microsoft Tong can help on easy task currently for the first question, #731 mentioned that $IsWindows is reserved name in PowerShell 6 version. Thanks!

juhlee-microsoft commented:
> @juhlee-microsoft Tong can help on easy task currently for the first question, #731 mentioned that $IsWindows is reserved name in PowerShell 6 version. Thanks!

Thanks, does it need to test before merging?

LiliDeng commented:
We will run a test to make sure though it is a simple.

LiliDeng commented on Libraries/Azure.psm1:
```diff
@@ -685,7 +685,7 @@ Function Get-AllDeploymentData($ResourceGroups)
 
 Function Get-NewVMName ($namePrefix, $numberOfVMs) {
-    if ($IsWindows) {
+    if ($IsWindowsImage) {
```
$global:IsWindowsImage

LiliDeng commented on Libraries/CommonFunctions.psm1:
```diff
@@ -305,7 +305,7 @@ function Is-VmAlive {
         $retryCount += 1
         foreach ( $vm in $AllVMDataObject) {
-            if ($IsWindows) {
+            if ($IsWindowsImage) {
```
$global:IsWindowsImage

LiliDeng commented on Libraries/CommonFunctions.psm1:
```diff
@@ -317,7 +317,7 @@ function Is-VmAlive {
                 # Note(v-advlad): Check for kernel panic once every ${kernelPanicPeriod} retries on Linux Azure
                 if (($retryCount % $kernelPanicPeriod -eq 0) -and ($TestPlatform -eq "Azure") `
-                    -and (!$IsWindows) -and (Check-AzureVmKernelPanic $vm)) {
+                    -and (!$IsWindowsImage) -and (Check-AzureVmKernelPanic $vm)) {
```
！$global:IsWindowsImage

LiliDeng commented on Libraries/HyperV.psm1:
```diff
@@ -483,7 +483,7 @@ function Get-AllHyperVDeployementData($HyperVGroupNames,$GlobalConfig,$RetryCoun
         Add-Member -InputObject $objNode -MemberType NoteProperty -Name InternalIP -Value $null -Force
         Add-Member -InputObject $objNode -MemberType NoteProperty -Name RoleName -Value $null -Force
-        if ($IsWindows) {
+        if ($IsWindowsImage) {
```
$global:IsWindowsImage

LiliDeng commented on Libraries/HyperV.psm1:
```diff
@@ -549,7 +549,7 @@ Function Inject-HostnamesInHyperVVMs($allVMData)
         {
             Write-LogInfo "Injecting hostname '$($VM.RoleName)' in HyperV VM..."
-            if (!$IsWindows) {
+            if (!$IsWindowsImage) {
```
！$global:IsWindowsImage

LiliDeng commented on Libraries/HyperV.psm1:
```diff
@@ -691,7 +691,7 @@ function Check-IP {
                 if ($vmIP) {
                     $vmIP = $([ipaddress]$vmIP.trim()).IPAddressToString
-                    if ($IsWindows) {
+                    if ($IsWindowsImage) {
```
$global:IsWindowsImage

LiliDeng commented on TestControllers/TestController.psm1:
```diff
@@ -384,13 +384,13 @@ Class TestController
 			$this.TestProvider.RunSetup($VmData, $CurrentTestData, $testParameters, $ApplyCheckpoint)
 
-			if (!$this.IsWindows) {
+			if (!$this.IsWindowsImage) {
```
!$global:IsWindowsImage

LiliDeng commented on TestControllers/TestController.psm1:
```diff
@@ -384,13 +384,13 @@ Class TestController
 			# Upload test files to VMs
 			if ($CurrentTestData.files) {
-				if(!$this.IsWindows){
+				if(!$this.IsWindowsImage){
```
!$global:IsWindowsImage

LiliDeng commented on TestControllers/TestController.psm1:
```diff
@@ -430,14 +430,14 @@ Class TestController
 
 		# Do log collecting and VM clean up
-		if (!$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True") {
+		if (!$this.IsWindowsImage -and $testParameters["SkipVerifyKernelLogs"] -ne "True") {
```
!$global:IsWindowsImage

LiliDeng commented on TestControllers/TestController.psm1:
```diff
@@ -430,14 +430,14 @@ Class TestController
 
-		$collectDetailLogs = !$this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
-		$doRemoveFiles = $this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !($this.ResourceCleanup -imatch "Keep") -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
+		$collectDetailLogs = !$this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !$this.IsWindowsImage -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
```
!$global:IsWindowsImage

LiliDeng commented on TestControllers/TestController.psm1:
```diff
@@ -430,14 +430,14 @@ Class TestController
-		$collectDetailLogs = !$this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
-		$doRemoveFiles = $this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !($this.ResourceCleanup -imatch "Keep") -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
+		$collectDetailLogs = !$this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !$this.IsWindowsImage -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
+		$doRemoveFiles = $this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !($this.ResourceCleanup -imatch "Keep") -and !$this.IsWindowsImage -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
```
!$global:IsWindowsImage

LiliDeng commented:
@lizzha Please help merge this PR, testing pass. Final result failed for the same issue which #739 fixed, I will create a new PR for such issue.

02/19/2019 06:24:13 : [INFO ] 	Connections=1 : throughput=3.98Gbps cyclePerBytet=2.08 Avg_TCP_lat=1742.749 <br />
02/19/2019 06:24:13 : [INFO ] Uploading the test results..
02/19/2019 06:24:13 : [INFO ] Database details are not provided. Results will not be uploaded to database!
02/19/2019 06:24:13 : [INFO ] Test result : PASS

juhlee-microsoft committed:
Change from Aborted to Skipped

Log said "skipped" but actually "aborted" inconsistenly.

juhlee-microsoft committed:
Added filter out for Ubuntu 12.04

The feature enable patch was not checked into this version of kernel source code. kernel.ubuntu.com/ubuntu/ubuntu-precise.git

juhlee-microsoft committed:
Merge branch 'master' of https://github.com/LIS/LISAv2

juhlee-microsoft committed:
Update for legacy OS support
- Added DISTRO ret value for Ubuntu 12.04
- Update ETHTOOL-CHECK-CHANNEL.sh
- Update How-to-use.md

lubaihua33 reviewed (commented):
Do we need to fix in master branch?
